server/audit/permissions.py

```diff
@@ -44,3 +44,28 @@ def has_permission(self, request, view):
             user = CustomUser.objects.get(id=request.data['customuser'])
             return user.organization_id == request.user.organization_id
         return True
+
+
+class IsAssignedSK(BasePermission):
+    message = "Must be a Stock Keeper that is retrieving their own audits"
+
+    def has_permission(self, request, view):
+        try:
+            user = CustomUser.objects.get(email=request.user)
+            if request.method == 'GET':
+                assigned_sk = (request.query_params.get('assigned_sk')
+                or request.query_params.get('customuser_id'))
+                return user.role == 'SK' and str(assigned_sk) == str(user.id)
+
+            if request.method == 'DELETE':
+                record_id = view.kwargs["pk"]
+                record = Record.objects.get(pk=record_id)
+                assigned_sk = record.audit.assigned_sk.get(id=user.id)
+                return user.role == 'SK' and assigned_sk
+
+            if request.data['audit']:
+                audit = Audit.objects.get(audit_id=request.data['audit'])
+                assigned_sk = audit.assigned_sk.get(id=user.id)
+            return user.role == 'SK' and assigned_sk
+        except (ObjectDoesNotExist, KeyError):
+            return False
```

Comment:
I see the same check for whether the role is an SK. Can't we just put that in a different permission?
The goal is to make each permission as easy to understand as possible by making them small and digestible, so I think this permission alone could probably be broken up into 4, but I will let that decision go to you, but let's at least extract the check for whether the role is an SK.

Reply:
yes i agree the permissions need refactoring

Reply:
It seems this is the majority of this PR is the permissions, this will be taken care of